datafusion/expr-common/src/type_coercion/binary.rs

```diff
@@ -1607,6 +1629,18 @@ mod tests {
         }};
     }
 
+    /// Test coercion rules for assymetric binary operators
+    ///
+    /// Applies coercion rules for `$LHS_TYPE $OP $RHS_TYPE` and asserts that the
+    /// the result types are `$RESULT_TYPE1` and `$RESULT_TYPE2` respectively
+    macro_rules! test_coercion_assymetric_binary_rule {
```

Comment:
What's the benefit of this being a macro? Could this be ordinary function?

Reply:
I don't know -- I just copied how the rest is done here, via macros too. This macro is a clone of test_coercion_binary_rule macro.

Reply:
i assumed so. But also -- macros aren't "better functions", they come with downsides, so if something is expressible as a function, it should.